### ISSUE TITLE
test_basics:test_idempotent non-deterministically fails 

### ISSUE THREAD

**Issue by lamby** [mozilla/bleach#161]
Originally reported here: https://bugs.debian.org/798441

```
$ while nosetests3 bleach.tests.test_basics:test_idempotent; do :; done
```

This eventually results in https://gist.github.com/lamby/bacb42d3fffde936d4a6

Curiously it only occurs under Python 3, ie. not if we `s/nosetests3/nosetests/`

**Comment by jsocol** [comment]
It's probably because the dict iteration order is non-deterministic. This was fixed in clean in #137, (issue #121). I'm _guessing_ that the fact that it takes a while to fail is that Python randomizes certain seed values on start-up to prevent [hashtable collision attacks](http://bugs.python.org/issue13703), and once in a while, those values cause it to return the attributes in a different order.

**Comment by lamby** [comment]
> It's probably because the dict iteration order is non-deterministic.

Yep. We see this a lot in the Reproducible Builds project. Glad there's a fix. :)
